src/server/stat_job.c

```diff
@@ -378,12 +378,15 @@ status_subjob(job *pjob, struct batch_request *preq, svrattrl *pal, int subj, pb
 	if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_ArrayJob) == 0)
 		return PBSE_IVALREQ;
 
-	/* if subjob is running, use real job structure */
+	/* if subjob job obj exists, use real job structure */
 
-	if (get_subjob_state(pjob, subj) == JOB_STATE_RUNNING) {
-		psubjob = find_job(mk_subjob_id(pjob, subj));
-		if (psubjob)
-			status_job(psubjob, preq, pal, pstathd, bad);
+	psubjob = find_job(mk_subjob_id(pjob, subj));
```

Comment:
find_job() here is required only when subjob is in RUNNING state. Please change the code accordingly.

Reply:
Fixed in rc 25 https://github.com/PBSPro/pbspro/pull/590/commits/f5c0e45e72a2ec7ef7af6513cb969adeeb697a2f